lib/resty/upstream/healthcheck.lua

```diff
@@ -134,7 +138,15 @@ local function peer_fail(ctx, is_backup, id, peer)
     -- print("ctx fall: ", ctx.fall, ", peer down: ", peer.down,
           -- ", fails: ", fails)
 
-    if not peer.down and fails >= ctx.fall then
+    local u_key = gen_upstream_key(u,"all_down")
+
+    local all_down, err = dict:get(u_key)
+    if not all_down then
+        errlog("Failed to get all down flag for upstream " .. u .. ". ", err)
+        return
+    end
+
+    if not peer.down and fails >= ctx.fall and all_down ~= 1 then
```

Comment:
I think the `all_down` flag should be a boolean instead of a number. The lua shared dict does support boolean typed values.

Reply:
Also thought about this. But, ngx.shared.DICT.get returns nil if key does not exists. So, all checks like "if not key"  after each dict:get() have to be adapted then. That's why I chose {0,1}

I mean that "if not key" will be true for both false and nill values.